dali/pipeline/operators/color/brightness_contrast.h

```diff
@@ -0,0 +1,193 @@
+using BrightnessContrastKernel = typename Kernel<Backend, OutputType, InputType, ndims>::type;
+
+
+/**
+ * Assign argument, whether it is a single value (for sample-wise processing)
+ * or vector of values (for batch processing, where every argument is defined per-sample)
+ */
+template <typename Backend>
+void assign_argument_value(const OpSpec &, const std::string &, argument_t<Backend> &) {
+  DALI_FAIL("Unsupported Backend. You may want to write your own specialization.");
+}
+
+
+template <>
+void assign_argument_value<CPUBackend>(const OpSpec &spec, const std::string &arg_name,
+                                       argument_t<CPUBackend> &arg) {
+  std::vector<float> tmp;
+  GetSingleOrRepeatedArg(spec, tmp, arg_name);
+  arg = tmp[0];
+}
+
+
+template <>
+void assign_argument_value<GPUBackend>(const OpSpec &spec, const std::string &arg_name,
+                                       argument_t<GPUBackend> &arg) {
+  GetSingleOrRepeatedArg(spec, arg, arg_name);
+}
+
+}  // namespace detail
+
+
+template <typename Backend>
+class BrightnessContrast : public Operator<Backend> {
+ public:
+  explicit BrightnessContrast(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          output_type_(spec.GetArgument<DALIDataType>(detail::kOutputType)) {
+    detail::assign_argument_value<Backend>(spec, detail::kBrightness, brightness_);
+    detail::assign_argument_value<Backend>(spec, detail::kContrast, contrast_);
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
+    }
+  }
+
+
+  ~BrightnessContrast() = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(BrightnessContrast);
+
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const ::dali::workspace_t<Backend> &ws) override {
+    const auto &input = ws.template InputRef<Backend>(0);
+    const auto &output = ws.template OutputRef<Backend>(0);
+    output_desc.resize(1);
+
+    TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+        {
+          TypeInfo type;
+          type.SetType<OutputType>(output_type_);
+          output_desc[0] = {input.shape(), type};
+        }
+    ), DALI_FAIL("Unsupported image type"))  // NOLINT
+    return true;
+  }
+
+  /**
+   * So that compiler wouldn't complain, that
+   * "overloaded virtual function `dali::Operator<dali::CPUBackend>::RunImpl` is only partially
+   * overridden in class `dali::brightness_contrast::BrightnessContrast<dali::CPUBackend>`"
+   */
+  using Operator<Backend>::RunImpl;
+
+
+  void RunImpl(Workspace<Backend> &ws) {
+    const auto &input = ws.template Input<Backend>(0);
+    auto &output = ws.template Output<Backend>(0);
+
+    TYPE_SWITCH(input.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
```

Comment:
DALI_TYPE_SWITCH_WITH_FP16 ?

Reply:
Regarding types, that are supported here, I followed [requirements](https://jirasw.nvidia.com/browse/DALI-821), so I'd like to manually specify types, that are supported

Reply:
It says:
```
Output types: [u8, i8, u16, i16, u32, i32, f32]
Input types: [u8, i8, u16, i16, u32, i32, f32]
```
you are not covering all the cases.

In my opinion, there is no reason for not using the DALI_TYPE_SWITCH_WITH_FP16 here and cover all available types